From 7c1f4284562e905372b5c14b6017f1eb438f54b5 Mon Sep 17 00:00:00 2001
From: Jan-Marek Glogowski <glogow@fbihome.de>
Date: Thu, 26 Aug 2021 14:23:29 +0200
Subject: tdf#144037 Qt5 just close popup with mouse outside

How hard could it be to close a popup window when losing focus or
clicking outside of it (AKA tdf#143114 problem 3)? At least with
Qt it feels like solving an NP problem...

So don't always force-close the popup window if clicked anywhere
on it. Seems to be consistent with gen and gtk3.

Change-Id: Ia1c231c0b51a55bec2c587feffcf14bb4b4cee55
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/121093
Tested-by: Jenkins
Reviewed-by: Jan-Marek Glogowski <glogow@fbihome.de>
(cherry picked from commit a1a83b6b3b4459de353e0cdc8cdf402370c0a5c4)
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/121064
Reviewed-by: Adolfo Jayme Barrientos <fitojb@ubuntu.com>
---
 vcl/qt5/Qt5Widget.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vcl/qt5/Qt5Widget.cxx b/vcl/qt5/Qt5Widget.cxx
index 864701340ad3..64ad4f7539c4 100644
--- a/vcl/qt5/Qt5Widget.cxx
+++ b/vcl/qt5/Qt5Widget.cxx
@@ -182,7 +182,7 @@ void Qt5Widget::mousePressEvent(QMouseEvent* pEvent)
 {
     handleMousePressEvent(m_rFrame, pEvent);
     if (m_rFrame.isPopup()
-        || !geometry().translated(geometry().topLeft() * -1).contains(pEvent->pos()))
+        && !geometry().translated(geometry().topLeft() * -1).contains(pEvent->pos()))
         closePopup();
 }
 
-- 
cgit v1.2.1

